### PR TITLE
Speedups

### PR DESCRIPTION
**Pull request by sebastianGehrmann**
Introduces various parallelism and caching methods to speed up running and (re)running the script. 

Additionally (mostly) addresses the following issues:

#38 
#12 
#44 
#43 

Not ready for merge yet, still need to refactor.

### CODE REVIEW

**Comment by sebastianGehrmann** [comment]
@tuetschek You wanna have a look at the new design now? I think it's ready for merging.

File: gem_metrics/texts.py [hunk 1]
```diff
@@ -13,72 +15,119 @@ class Texts:
...
-
-        # allow bare lists of strings as well as lists of dicts
-        if self.all_data and isinstance(self.all_data[0], str):
+            self.filename = data.get("filename")
+            self.all_data = data["values"]
+        logger.info(
+            f"Loading {self.__class__.__name__.lower()} for {str(self.filename)}"
+        )
+
+        self.language = languages.get(alpha_2=language)
```
Comment (tuetschek) [inline gem_metrics/texts.py:33]: Any specific reason to remove the flexibility here (having language="English" worked the same as "eng" and "en", now it doesn't)?

File: gem_metrics/__init__.py [hunk 2]
```diff
@@ -112,84 +198,94 @@ def process_submission(outs: Submission, refs: Optional[Dict], srcs: Optional[Di
...
-        srcs_ds = srcs[dataset] if (srcs and dataset in srcs) else load_sources(dataset)
-        values[dataset] = compute(outs_ds, refs_ds, srcs_ds, metrics_dict=metrics_dict)
-    return values
+        refs_ds = refs.get(dataset, None)
+        srcs_ds = srcs.get(dataset, None)
+        job_args.append(
+            (dataset, outs_ds, refs_ds, srcs_ds, parallel_metric_dict, cache)
+        )
+
+    pool = Pool(processes=12)
```
Comment (tuetschek) [inline gem_metrics/__init__.py:223]: What would this do with much fewer cores? How hard would it be to parameterize it?

**Comment by tuetschek** [comment]
@sebastianGehrmann It's a bit hard to read with the formatting changes, but just by looking at the code, everything seems good to me! I just left 2 tiny comments (see above)

File: gem_metrics/texts.py (hunk 1 above)
Comment (sebastianGehrmann) [reply gem_metrics/texts.py:33]: Now it is set globally in the `config.py` and is no longer required in the individual submission files. Hence the simplification.

File: gem_metrics/__init__.py (hunk 2 above)
Comment (sebastianGehrmann) [reply gem_metrics/__init__.py:223]: I can add a parameter for this. In theory it should work with <12 cores as well though.

**Comment by sebastianGehrmann** [comment]
Thanks! Sorry for the formatting, I have an automatic linter that does that sometimes.

**Comment by sebastianGehrmann** [comment]
Added the cmd arg for number of threads.

File: gem_metrics/texts.py (hunk 1 above)
Comment (tuetschek) [reply gem_metrics/texts.py:33]: OK, makes sense :-)

**Comment by tuetschek** [comment]
Thanks for adding that! Good to merge for me 🙂

**Comment by sebastianGehrmann** [comment]
Awesome, great to have this out of the way :)

**Pull request merged**